src/operations/blas1_trees.hpp

```diff
@@ -175,6 +176,15 @@ SYCL_BLAS_INLINE bool Assign<lhs_t, rhs_t>::valid_thread(
           Assign<lhs_t, rhs_t>::get_size());
 }
 
+template <typename lhs_t, typename rhs_t>
+template <int vector_size>
+SYCL_BLAS_INLINE bool Assign<lhs_t, rhs_t>::valid_thread(
+    cl::sycl::nd_item<1> ndItem) const {
+  using index_t = typename Assign<lhs_t, rhs_t>::index_t;
+  return (static_cast<index_t>(ndItem.get_global_id(0)) <
+          (Assign<lhs_t, rhs_t>::get_size() - 1 / vector_size) + 1);
```

Comment:
Did you mean `( (Assign<lhs_t, rhs_t>::get_size() - 1)/ vector_size) + 1)` here ?